tests/eos-test-utils.c

```diff
@@ -351,6 +351,140 @@ prepare_sysroot_contents (GFile *repo,
 }
 
 static gboolean
+generate_big_file_for_delta_update (GFile *all_commits_dir,
+                                    guint commit_no,
+                                    GError **error)
+{
+  gsize byte_count = 10 * 1024 * 1024 + 1;
+  g_autofree gchar *data = g_malloc (byte_count);
+  g_autoptr(GFile) big_file = NULL;
+  g_autoptr(GBytes) contents = NULL;
+
+  memset (data, 'x', byte_count);
+  data[byte_count / 2] = 'a' + commit_no;
+  big_file = g_file_get_child (all_commits_dir, "bigfile");
+  contents = g_bytes_new_take (g_steal_pointer (&data), byte_count);
+
+  return create_file (big_file, contents, error);
+}
+
+static gboolean
+fill_all_commits_dir (GFile *all_commits_dir,
+                      guint commit_no,
+                      GError **error)
+{
+  const gchar *dirnames[] = { "a", "b", "c" };
+  const gchar *filenames[] = { "x", "y", "z" };
+  guint iter;
+
+  {
+    g_autofree gchar *commit_dirname = g_strdup_printf ("commit%u.dir", commit_no);
+    g_autoptr(GFile) commit_dir = g_file_get_child (all_commits_dir, commit_dirname);
+    if (!create_directory (commit_dir, error))
+      return FALSE;
+  }
+
+  for (iter = 0; iter <= commit_no; ++iter)
```

Comment:
That `<10` check is in another function though, right? Generally the boundary for precondition checks is at the function boundary.